src/ScottPlot/Drawing/GDI.cs

```diff
@@ -18,6 +18,34 @@ public static class GDI
         private const float xMultiplierMacOS = 82.82f / 72;
         private const float yMultiplierMacOS = 27.16f / 20;
 
+        /// <summary>
+        /// Test if the system can properly render graphics using System.Drawing.Common.
+        /// If an exception occurs this will return a string description, otherwise it returns null.
+        /// </summary>
+        public static string DrawingTest()
+        {
+            try
+            {
+                var bmp = new Bitmap(10, 10);
+                using var gfx = GDI.Graphics(bmp);
+                gfx.Clear(Color.Magenta);
+                gfx.DrawLine(Pens.Black, 0, 0, 10, 10);
+
+                using var stream = new System.IO.MemoryStream();
+                bmp.Save(stream, ImageFormat.Bmp);
+                byte[] bytes = stream.ToArray();
+                int expectedBitmapSize = 454;
+                if (bytes.Length != expectedBitmapSize)
+                    throw new InvalidOperationException($"test image was {bytes.Length} bytes (expected {expectedBitmapSize})");
```

Comment:
Throwing something in order to immediately catch it is something of an antipattern. It's using exception handling as flow control instead of an if/else.

This method should probably either return a boolean to indicate success/failure, or just let any exceptions bubble up to the caller.

If the actual error message is important then it should throw an exception which is caught by the caller and the error message surfaced to the user. Alternatively, you can return false to indicate failure and then set an error field somewhere (like in many C APIs), or return a struct with a success field set to false and some extra information (like in many AJAX libraries).

I don't normally like the C way of doing things because it's used everywhere to get around the lack of exception support, however it does have some benefits:
- It does not obligate the caller to handle any exceptions
    - Thus it's a pattern often used in AJAX when it doesn't matter if the request fails (or when failures are already handled by some middleware which will retry)
- Performance and compatibility with embedded systems (not really a consideration here)

Reply:
Returning a struct like this would probably be the best option, provided you don't want to let the exception bubble up to the caller. That choice depends on whether you think the caller ought to be obligated to handle the issue or die.

```cs
struct TestResult {
    bool success;
    string errorMessage;
}
```